src/main/java/com/thullo/service/BoardServiceImpl.java

```diff
@@ -153,6 +129,30 @@ public List<BoardResponse> getBoards(UserPrincipal userPrincipal) throws UserExc
         return boardResponses;
     }
 
+    @Override
+    public BoardResponse updateBoard(UpdateBoardRequest boardRequest, UserPrincipal userPrincipal)
+            throws UserException, BadRequestException, IOException {
+
+        User user = findByEmail(userPrincipal.getEmail());
+        Board board = getBoardInternal(boardRequest.getBoardTag());
+
+        if (board == null) throw new BadRequestException(BOARD_NOT_FOUND);
+        List<Board> userBoards =  user.getBoards();
+        String imageUrl;
+
+        for (Board userboard : userBoards){
+            if(userboard == board){
+                if(boardRequest.getName() != null) userboard.setName(boardRequest.getName());
```

Comment:
Instead of using if checks, use modelMapper and add the other field that modelMapper can't handle manually

Reply:
I used a manual check here because I noticed when I used model mapper, it sets some fields that were not requested updates to null. So I had to write the manual check to ensure the field is not null before I override the previous value, perhaps my implementation was bad, please kindly guide me through using the model mapper more effectively.

Thanks.
Lekan.